Mathlib/Data/Finsupp/Pointwise.lean

```diff
@@ -111,4 +111,24 @@ theorem coe_pointwise_smul [Semiring β] (f : α → β) (g : α →₀ β) : 
 instance pointwiseModule [Semiring β] : Module (α → β) (α →₀ β) :=
   Function.Injective.module _ coeFnAddHom DFunLike.coe_injective coe_pointwise_smul
 
+section
+
+variable [Semiring β] [AddCommMonoid γ] [Module β γ]
+
+instance pointwiseModuleScalar : HMul (α →₀ β) (α → γ) (α →₀ γ) where
```

Comment:
I think inheriting a `*` from a `smul` in this way is not an established pattern in mathlib. This would maybe be a _little_ less objectionable if it used `HSMul`, but fundamentally the heterogenous classes create nasty diamond issues (which I touch on briefly in [Section 4.8](https://eric-wieser.github.io/thesis/eric-wieser-thesis.pdf#page=63) of my thesis.

Reply:
Presumably I should make it a `def` instead? https://github.com/leanprover-community/mathlib4/pull/18578/files#diff-b2e63ded24baf2b6dbf29bf1c8d5ff01b0db9b4201ce0726503848038e25b241R62